lib/server.js

```diff
@@ -1143,19 +1143,37 @@ Server.prototype._setupRequest = function _setupRequest(req, res) {
     // the connection (or we lose the connection).
     // we consider a closed request as flushed from metrics point of view
     function onClose() {
-        res._flushed = true;
-        req._timeFlushed = process.hrtime();
-
         res.removeListener('finish', onFinish);
         res.removeListener('error', onError);
+
         req._connectionState = 'close';
 
+        // Do no handle close if request is already flushed
+        // In Node >= 10 res's close event is always emitted
```

Comment:
Is is Node > 10.4.0 at least? It seems that change hasn't been made it to any release yet.